production/helm/loki/templates/monitoring/grafana-agent.yaml

```diff
@@ -18,6 +18,10 @@ spec:
   serviceAccountName: {{ include "loki.fullname" $ }}-grafana-agent
   enableConfigReadAPI: {{ .enableConfigReadAPI }}
   logs:
+    instanceSelector:
+      matchLabels:
+        {{- include "loki.selectorLabels" $ | nindent 8 }}
+  metrics:
```

Comment:
shouldn't we only include the metrics section if a metrics instance has been defined?

Reply:
``
{{- with .Values.monitoring.serviceMonitor.metricsInstance }}
```
at the top should do this, shouldn't it?